app/actions/CommonActions.scala

```diff
@@ -22,17 +27,43 @@ object CommonActions {
 
   object PreReleaseFeature extends ActionBuilder[Request] with LazyLogging {
     override def invokeBlock[A](request: Request[A], block: (Request[A]) => Future[Result]): Future[Result] = {
-      val cookie = request.cookies.get(qaCookie.name)
-
-      if (cookie.exists(_.value == qaCookie.value)) {
+      val qaCookie = request.cookies.get(passthroughCookie.name)
+      if (qaCookie.exists(_.value == passthroughCookie.value)) {
         block(request)
       } else {
-        cookie.foreach(_ => logger.warn("Invalid QA Cookie supplied"))
+        qaCookie.foreach(_ => logger.warn("Invalid QA Cookie supplied"))
         OAuthActions.AuthAction.authenticate(request, block)
       }
     }
   }
 
+  class RequestWithCookies[A](request: Request[A]) extends WrappedRequest[A](request) {
+    val testUserCookie = request.cookies.get(testUserCookieName)
```

Comment:
I am afraid we have totally overlooked CAS proxy while implementing this. I would park it for today as I need a bit more context in order to go ahead. Hopefully we can look into this together tomorrow.